### ISSUE TITLE
Crashing at random time

### ISSUE THREAD

**Issue by benbenlaw** [AllTheMods/ATM-6#42]
Seems to be a problem with forbidden arcanus, this is the latest update from today

**Comment by benbenlaw** [comment]
[crash-2020-08-06_14.04.24-server.txt](https://github.com/AllTheMods/ATM-6/files/5034954/crash-2020-08-06_14.04.24-server.txt)

**Comment by surfersmurf** [comment]
Hello! I have the same issue.

https://pastebin.com/az8Y9004

**Comment by benbenlaw** [comment]
Yeah its really anoying literall cant play

**Comment by LeadAssimilator** [comment]
I don't believe this is a Forbidden Arcanus bug actually, but is instead an issue with Minecraft's Vines.  The updated Forbidden Arcanus renamed the Dark Stone block id, and Quark's Cave Roots which are based on Minecraft's Vines, crash when they attempt to continue their growth on and over the now defunct blocks.  This basically can happen any time already placed blocks that have vines on them get removed due to rename, mod removal or disable since Forge replaces them with air.  Quark might be able to workaround the crash for Cave Roots, but the core Minecraft Vines would still cause problems so Mojang will probably have to fix this.

Unless there is a Forge config to replace defunct blocks with something other than air, the only other option is to use a world editor to replace all forbidden_arcanus:dark_stone blocks with forbidden_arcanus:darkstone.  You could also painfully do this in game via /gamemode spectator and the /fill <from pos> <to pos> forbidden_arcanus:dark_stone replace forbidden_arcanus:darkstone command.

Although I don't recommend it, you could also break all of the vines and cave roots in the problem chunks that are not directly anchored to a block, as that might be where a dark_stone once was, but to do that outside of spectator mode, you have to do /gamerule randomTickSpeed 0 to disable vine and cave root growth so you can enter the chunk without it crashing, then reset it back to 3 when you are done.  You can also boost it to 1000 to test if you fixed it - that will cause vines to grow super fast and trigger the crash if you didn't get them all.

**Comment by surfersmurf** [comment]
Thank you very much for your insight and possible solutions!

Do you think removing the mod, then loading up the save (so all dark stone blocks are removed from the world) and then reinstalling the mod so dark stone is generated in new chunks would also work?

**Comment by LeadAssimilator** [comment]
No.  Removing a mod does not actually remove its blocks.  As stated above, removing, or disabling a mod will in fact cause this crash to happen when vines or cave roots try to grow over or on the blocks that are no longer there because they are in fact still there, but replaced with air by Forge - the air block does not have north,south,east,west boolean properties!  The only way I know to fix this is to replace all the defunct blocks in every chunk with something else.

There might be easier methods, but what worked for me was using MCA Selector to delete all chunks outside of my base to fix worldgen given the major updates in 0.24b.  Then I installed the latest master build of WorldEdit #5217 from here https://builds.enginehub.org/job/worldedit?branch=master (the latest stable on Curse from last month hangs).  I started the server, set the randomTickSpeed to 0 asap, and joined the world.  I then used WorldEdit //chunk, //expand and //replace commands to replace forbidden_arcanus:dark_stone with forbidden_arcanus:darkstone and finally reset randomTickSpeed to 3.
